python/setup.py

```diff
@@ -198,7 +199,7 @@ def get_packages(self):
         ],
     }
 
-    if os.getenv("RAY_EXTRA_CPP") == "1":
+    if RAY_EXTRA_CPP:
```

Comment:
IMO we should check here if the `setup_spec.version` is a proper version (like 1.6.0) or the development version. We will not upload development cpp wheels to pypi. Thus, when installing a dev version from a specific commit, installing `ray[all]` afterwards will always downgrade ray to the latest release.

Reply:
So to be clear, if `setup_spec.version == "2.0.0.dev0"` then we want to not require a specific `ray_cpp` version or default to not including `ray_cpp` in extras

Reply:
Maybe we will rename "2.0.0.dev0" to "3.0.0.dev0" later?  Decide the logic by `if "dev" in setup_spec.version`?